infrastructure/simulation-tool/src/operations/common.ts

```diff
@@ -33,6 +33,38 @@ const ensureRollupFunds = async (totalDepositAmount: BigNumber, l2Wallet: Rollup
     }
 };
 
+const ensureRollupFundsFromRollup = async (
+    totalDepositAmount: BigNumber,
+    from: RollupWallet,
+    l2Wallet: RollupWallet
+) => {
+    if (totalDepositAmount.gt(await l2Wallet.getBalance('RBTC'))) {
+        const transfer = await from.syncTransfer({
+            to: l2Wallet.address(),
+            amount: utils.closestGreaterOrEqPackableTransactionAmount(totalDepositAmount),
+            token: 'RBTC',
+            nonce: 'committed'
+        });
+
+        await transfer.awaitReceipt();
+    }
+};
+
+const ensureL2AccountActivation = async (account: RollupWallet) => {
+    const isActive = await account.isSigningKeySet();
```

Comment:
```suggestion
    const isActive = await account.isSigningKeySet() && Boolean(await user.getAccountId());
```

also an account on L2 is only activated upon transaction to it, so to ensure activation you'd also deposit/transfer.